### PR TITLE
[WIP] Making right endings

### PR DESCRIPTION
**Pull request by PodviaznikovPavlo**
@podviaznikov 
Have made endings replace on test data. NOT for all variants. What would you say?
If its ok, i`ll continue switch. There also must be 'if' statement for countries such (Сполучені Шатти Америки, Велика Британія). I will do it too.

### CODE REVIEW

**Comment by PodviaznikovPavlo** [comment]
@podviaznikov 
Well, for such countries we can hardcode it.

**Comment by podviaznikov** [comment]
We can't do any if. We need very generic code, because we will not have just countries, cities etc.

We can ignore such countries. Do you know what is the rule? You can't create correct form for the countries that have several words in the name?

File: app.js [hunk 1]
```diff
@@ -7,10 +7,16 @@ var express = require('express'),
 
 var questions = [];
 
+var originCountries = ['Україна', 'Англія', 'Франція', 'Іспанія', 'Німеччина', 'Чехія', 'Молдова', 'Уельс', 'Китай', 'Азербайджан'];
```
Comment (podviaznikov) [inline app.js]: can you put code related to questions in the file question.js?

Comment (PodviaznikovPavlo) [inline app.js]: Yes, i did it. Create questions.js. It works.

**Comment by PodviaznikovPavlo** [comment]
@podviaznikov 
I dont understand. About what rules you are talking? 
So my way is wrong? Using switch how i use.

**Comment by podviaznikov** [comment]
Don't need switches or ifs or any hardcodes. We are going to generate questions not just for countries. We need 80000 questions. Will you hard code exceptions for that many types of questions?

If the name of the country has more than one word - just remove it/ignore it. Clear?

File: app.js [hunk 2]
```diff
@@ -1,17 +1,10 @@
...
 
-var questions = [];
-
-fs.readFile('./public/freebase_countries.json', 'utf8', function(err, data){
-    jsonData = JSON5.parse(data);
-    getDataFromJson(jsonData, questions);
-    console.log(questions);
-});
+var questionsList = questionsJS.correctQuestions();
+console.log(questionsList);
```
Comment (podviaznikov) [inline app.js:7]: why do you print countries in app.js? I didn't ask you to do it.

Leave questions only in questions.js.

Comment (PodviaznikovPavlo) [inline app.js:7]: This is for me. To see is it works.

Comment (podviaznikov) [inline app.js:7]: if you want to see it works you can add this console.log to questions.js and run it as `node questions.js`. Don't need to pollute main file.

**Comment by PodviaznikovPavlo** [comment]
@podviaznikov 

Okay, but i havent any ideas, how to change endings. 
Есть одна идея (я так напишу). Написать метод родительного падежа. Получается он получает слово в именительном падеже и возвращает в родительном. НУжно только посмотреть правила по которым происходится преобразование. Но вот сейчас задумался и все равно не знаю, получится ли. 
У тебя есть какие то идеи?

**Comment by podviaznikov** [comment]
That was I was telling you today, no? Read the rules for http://uk.wikipedia.org/wiki/%D0%92%D1%96%D0%B4%D0%BC%D1%96%D0%BD%D0%BE%D0%BA and write the function.

**Comment by podviaznikov** [comment]
And test that it works for most variants.

File: questions.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+//fs.readFile('./public/freebase_countries.json', 'utf8', function(err, data){
+//    jsonData = JSON5.parse(data);
+//    getDataFromJson(jsonData, questions);
+//    console.log(questionsJS.questions);
+//});
+
+for(var i in originCountries) {
+    correctNameEnding = changeCountryNameEnding(originCountries[i]);
+    questions.push(correctNameEnding);
+}
```
Comment (podviaznikov) [inline questions.js:19]: Can you read about `map` function and see if you can use it?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Comment (PodviaznikovPavlo) [inline questions.js:19]: I understand what is map. You mean we can make something like editCountries = originCountries.map(makeRightEnding());
and editCountries will be array with fixed data. yes?

Comment (podviaznikov) [inline questions.js:19]: Yes, I want you to read about map, understand how and when to use it.

Than I want you to change code and use map.
Your code is almost right (`editCountries = originCountries.map(makeRightEnding());`), but there is small error. Try to find it

Comment (PodviaznikovPavlo) [inline questions.js:19]: oh, i think thats way:

editCountries = originCountries.map(function(country){
    return makeRightEndings(country);
}); 

Am i right?

Comment (podviaznikov) [inline questions.js:19]: Yes, you are right. There is shorter version though:

```
editCountries = originCountries.map(makeRightEndings);
```

Comment (podviaznikov) [inline questions.js:19]: Did you understand the difference between three variants below?

```
editCountries = originCountries.map(makeRightEndings);
```

```
editCountries = originCountries.map(function(country){
return makeRightEndings(country);
});
```

and 

```
editCountries = originCountries.map(makeRightEnding());
```

Comment (PodviaznikovPavlo) [inline questions.js:19]: Yes.
1) its a put a function. makeRightEndings its whole function, we didnt call it here
2 and 3 variant are the same, but in 2 we create function inside the map
in 3 variand we create function and then put it on map and call it there. makeRightEndings will return something.

Comment (podviaznikov) [inline questions.js:19]: Actually your understanding (or explanation) of 3 cases is not correct.

2 and 3 are definitely not the same.

Please read https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map and revisit your comment if you can. Question remains the same.

Comment (PodviaznikovPavlo) [inline questions.js:19]: @podviaznikov 
okay, i'll read

**Pull request merged**